src/components/dashboard/EditClient.tsx

```diff
@@ -10,12 +10,22 @@ import { useDashboardClient } from '../pages/ClientDashboard';
 
 import EditRecord from '@/modules/form/components/EditRecord';
 import { Routes } from '@/routes/routes';
-import { ClientFieldsFragment, FormRole } from '@/types/gqlTypes';
+import {
+  ClientFieldsFragment,
+  FormRole,
+  useGetClientPermissionsQuery,
+} from '@/types/gqlTypes';
 import generateSafePath from '@/utils/generateSafePath';
 
 const Profile = () => {
   const { client } = useDashboardClient();
   const navigate = useNavigate();
+  const { data } = useGetClientPermissionsQuery({
+    variables: { id: client.id },
+  });
+
+  const { canViewDob, canViewFullSsn } = data?.client?.access || {};
```

Comment:
We should wait until the permissions are loaded until we render the `EditRecord` component